### PR TITLE
sql instance backup configuration enabled rule

### PR DESCRIPTION
**Pull request by doronafota**

### CODE REVIEW

File: cloudrail/knowledge/rules/gcp/non_context_aware/sql_database_instance_backup_configuration_enabled_rule.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    def get_id(self) -> str:
+        return 'non_car_cloud_sql_enable_backup'
+
+    def execute(self, env_context: GcpEnvironmentContext, parameters: Dict[ParameterType, any]) -> List[Issue]:
+        issues: List[Issue] = []
+        for sql_db_instance in env_context.sql_database_instances:
+            if not sql_db_instance.settings.backup_configuration.enabled:
+                issues.append(
+                    Issue(
+                        f"The Google Cloud database instance {sql_db_instance.get_type()} `{sql_db_instance.get_friendly_name()}` has backup_configured enabled flag not set.",
```
Comment (Imanuel-Mizrahi) [inline cloudrail/knowledge/rules/gcp/non_context_aware/sql_database_instance_backup_configuration_enabled_rule.py:19]: The "get_type" should return the "Google cloud database instance" so it sounds like this is duplicate...
Please check this

**Review by Imanuel-Mizrahi — CHANGES_REQUESTED** [review]
@doronafota , please merge your branch with the latest main, in order to make sure your changes does not impact the drifts

**Review by Imanuel-Mizrahi — APPROVED** [review]

File: cloudrail/knowledge/context/gcp/resources_builders/terraform/sql_database_instance_builder.py [hunk 2]
```diff
@@ -37,47 +39,52 @@ def build_settings_block(self, attributes: dict):
...
 
-        return GcpSqlDBInstanceSettingsIPConfig(True, None, None, [])
+        return GcpSqlDBInstanceSettingsIPConfig(ipv4_enabled, private_network, require_ssl, authorized_networks)
 
     def build_authorized_network(self, authorized_network: dict) -> GcpSqlDBInstanceIPConfigAuthNetworks:
         expiration_time = None
         if expiration_time_str := self._get_known_value(authorized_network, "expiration_time"):
             expiration_time_str_list = expiration_time_str.split(".")
             expiration_time_str = expiration_time_str_list[0] if len(expiration_time_str_list) > 1 else expiration_time_str_list[0][:-1]
-            expiration_time = datetime.strptime(expiration_time_str, '%Y-%m-%dT%H:%M:%S')
```
Comment (idan-aviv) [inline cloudrail/knowledge/context/gcp/resources_builders/terraform/sql_database_instance_builder.py:80]: same as above

File: cloudrail/knowledge/context/gcp/resources_builders/terraform/sql_database_instance_builder.py [hunk 3]
```diff
@@ -37,47 +39,52 @@ def build_settings_block(self, attributes: dict):
...
-        ip_configuration_block = self._get_known_value(settings, "ip_configuration")
-        ip_configuration = ip_configuration_block[0] if ip_configuration_block else None
-
-        if ip_configuration:
-            ipv4_enabled = self._get_known_value(ip_configuration, "ipv4_enabled", True)
-            private_network = ip_configuration.get("private_network")
-            require_ssl = self._get_known_value(ip_configuration, "require_ssl", False)
-            authorized_networks_list = self._get_known_value(ip_configuration, "authorized_networks")
-            authorized_networks = [self.build_authorized_network(authorized_network)
-                                   for authorized_network in authorized_networks_list] if authorized_networks_list else None
```
Comment (idan-aviv) [inline cloudrail/knowledge/context/gcp/resources_builders/terraform/sql_database_instance_builder.py:69]: same as above

File: cloudrail/knowledge/rules/gcp/non_context_aware/sql_database_instance_backup_configuration_enabled_rule.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+from cloudrail.knowledge.rules.rule_parameters.base_paramerter import ParameterType
+
+
+class SqlDatabaseBackupConfigurationEnabledRule(GcpBaseRule):
+    def get_id(self) -> str:
+        return 'non_car_cloud_sql_enable_backup'
+
+    def execute(self, env_context: GcpEnvironmentContext, parameters: Dict[ParameterType, any]) -> List[Issue]:
+        issues: List[Issue] = []
+        for sql_db_instance in env_context.sql_database_instances:
```
[the remark this answers is not in the document]
Comment (Imanuel-Mizrahi) [reply cloudrail/knowledge/rules/gcp/non_context_aware/sql_database_instance_backup_configuration_enabled_rule.py:15]: But @idan-aviv , following @doronafota latest changed in this PR, "settings" will never be empty.

**Review by Imanuel-Mizrahi — APPROVED** [review]

**Review by idan-aviv — APPROVED** [review]

File: cloudrail/knowledge/rules/gcp/non_context_aware/sql_database_instance_backup_configuration_enabled_rule.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (doronafota) [reply cloudrail/knowledge/rules/gcp/non_context_aware/sql_database_instance_backup_configuration_enabled_rule.py:15]: added checking if settings is not none

**Pull request merged**